### ISSUE TITLE
False positive on Null-coalescing operator within Conditional operator

### ISSUE THREAD

**Issue by DMDone** [SonarSource/sonar-dotnet#2299]
### Description

When I use Null-coalescing operator within Conditional operator I get a SonarQube warning.

### Repro steps

var x = a ?? false ? c : d;
^^^^^^^^^^^^^^^
               Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Expected behavior

No analyzer warning.

### Actual behavior

The part "false" is marked as Major Bug with message "Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed".
This is a wrong warning - priority of "??" is higher tnen "?:".

### Known workarounds

Using parentheses

### Related information

* SonarC# Version
* If running through the Scanner for MSBuild, its version

**Comment by andrei-epure-sonarsource** [comment]
Hi @DMDone and thanks for this issue.

Unfortunately, I cannot reproduce the FP you are mentioning.

When I try to build the below snippet I get a compiler error [CS0019](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0019) `CS0019: Operator '??' cannot be applied to operands of type 'string' and 'bool'` 

```
        void Foo(string a, bool b, string c, string d)
        {
            var x = a ?? b ? c : d;
        }
```

And with parantheses, as you say, there's no FP.

**Comment by DMDone** [comment]
Hi!

I apologize, my simplified example was not entirely correct.
B elow is a real example. D espite the FP warning, the code works correctly - if "items" is null then "subsequent code is executed".




>Среда, 17 апреля 2019, 13:09 +04:00 от Andrei Epure <notifications@github.com>:
>
>Hi  @DMDone and thanks for this issue.
>Unfortunately, I cannot reproduce the FP you are mentioning.
>When I try to build the below snippet I get a compiler error  CS0019 CS0019: Operator '??' cannot be applied to operands of type 'string' and 'bool'
>        void Foo(string a, bool b, string c, string d)
>        {
>            var x = a ?? b ? c : d;
>        }
>
>And with parantheses, as you say, there's no FP.
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  mute the thread .

Best regards,
Dmitry
